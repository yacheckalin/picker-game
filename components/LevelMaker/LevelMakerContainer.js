import React, { useState } from "react";

import LMForm from "./LMForm";
import LMGrid from "./LMGrid";

import { MIN_MAP_SIZE, DEFAULT_GRID_MAP } from "../../constants";
import shortid from "shortid";

const LevelMakerContainer = React.memo(() => {
  const [grid, setGrid] = useState(DEFAULT_GRID_MAP);
  const [size, setSize] = useState(MIN_MAP_SIZE);
  const [gridHash, setGridHash] = useState(shortid.generate());

  const handleGenerate = (size) => {
    // generate a matrix size x size
    let grid = [];
    for (let i = 0; i < size; i++) {
      grid[i] = new Array(size).fill(0);
    }
    setGrid(grid);
    setSize(size);
    setGridHash(shortid.generate());
  };

  return (
    <div className="col s12 card-panel lime lighten-4">
      <LMForm generate={handleGenerate} gridData={grid} />

      <LMGrid data={grid} size={size} key={gridHash} />
    </div>
  );
});

export default LevelMakerContainer;
