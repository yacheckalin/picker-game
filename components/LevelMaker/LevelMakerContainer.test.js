import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";

import LevelMakerContainer from "./LevelMakerContainer";

jest.mock("shortid", () => {
  return {
    generate: jest.fn(() => 1),
  };
});
describe("<LevelMakerContainer />", () => {
  test("renders and displays properly", () => {
    const wrapper = shallow(<LevelMakerContainer />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
