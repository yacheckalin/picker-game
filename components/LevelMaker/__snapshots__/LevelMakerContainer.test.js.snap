// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LevelMakerContainer /> renders and displays properly 1`] = `
<div
  className="col s12 card-panel lime lighten-4"
>
  <LMForm
    generate={[Function]}
    gridData={
      Array [
        Array [
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
        ],
      ]
    }
  />
  <LMGrid
    data={
      Array [
        Array [
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
        ],
      ]
    }
    key="1"
    size={3}
  />
</div>
`;
